From b6dc1ec9811e0750fc41c4af68be0f8cb1f51667 Mon Sep 17 00:00:00 2001
From: Gajanan Bhat <gajanan.bhat@intel.com>
Date: Mon, 19 Oct 2015 12:04:00 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: Avoid panel fitter calculations if
 it is being turned off

Avoid panel fitter calculations if mode is set to PFIT_OFF. Do not set any panel
fitter mode while turing it off.

	SQUASHME: This patch is to be squashed with the below patch
		commit 736a850003c9364569cad37a9e98afebefcfc207
		Author: Arun R Murthy <arun.r.murthy@intel.com>
		Date:   Tue Aug 11 14:29:54 2015 +0530
		SQUASHME! [VPG]: calculate the aspect ratio for pfit

Change-Id: I925de2893421af56d99433cd27283f354ad8b6fb
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-7392
Signed-off-by: Gajanan Bhat <gajanan.bhat@intel.com>
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 8beb93b..b6edfb0 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -11224,8 +11224,8 @@ int intel_set_disp_calc_flip(struct drm_mode_set_display *disp,
 			intel_crtc->scaling_src_size =
 				(((disp->panel_fitter.src_w - 1) << 16) |
 						(disp->panel_fitter.src_h - 1));
-
-			if (((mode->hdisplay * disp->panel_fitter.src_h) /
+			if (disp->panel_fitter.mode == PFIT_OFF);
+			else if (((mode->hdisplay * disp->panel_fitter.src_h) /
 						disp->panel_fitter.src_w) < mode->vdisplay)
 				pfit_mode |= PFIT_SCALING_LETTER;
 			else if (((mode->vdisplay * disp->panel_fitter.src_w) /
-- 
1.9.1

